packages/node/src/dev-server.ts

```diff
@@ -149,10 +150,22 @@ async function serializeRequest(message: IncomingMessage) {
 
 async function compileUserCode(entrypoint: string) {
   try {
-    const buildResult = await ncc(entrypoint, {
-      target: 'es2022',
+    const result = await esbuild.build({
+      platform: 'node',
+      target: 'node14',
+      sourcemap: 'inline',
+      bundle: true,
+      entryPoints: [entrypoint],
+      write: false, // operate in memory
+      format: 'cjs',
```

Comment:
I think we also need to detect ESM since we support ESM for production deployments with `@vercel/node`

Reply:
`format` here is output format. It does ingest ESM.

Reply:
The output from this is only when executing an edge function during `vc dev` inside the `edge-runtime`. The user never sees this output or has any way to use it directly.

I tested it in a new project where the `package.json` has `"type": "module"` to be sure and it does still execute edge functions properly.